internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,266 @@
+	err = json.Unmarshal(body, &res)
+	if err != nil {
+		return nil, fmt.Errorf("%s: parsing RPC response JSON", method)
+	}
+	return res, nil
+}
+
+func (t *transactionService) SignAndBuildNewFeeBumpTransaction(ctx context.Context, origTxXdr string) (*txnbuild.FeeBumpTransaction, error) {
+	genericTx, err := txnbuild.TransactionFromXDR(origTxXdr)
+	if err != nil {
+		return nil, tsserror.OriginalXDRMalformed
+	}
+	originalTx, txEmpty := genericTx.Transaction()
+	if !txEmpty {
+		return nil, tsserror.OriginalXDRMalformed
+	}
+	channelAccountPublicKey, err := t.ChannelAccountSignatureClient.GetAccountPublicKey(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting channel account public key: %w", err)
+	}
+	channelAccount, err := t.HorizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: channelAccountPublicKey})
+	if err != nil {
+		return nil, fmt.Errorf("getting channel account details from horizon: %w", err)
+	}
+	tx, err := txnbuild.NewTransaction(
+		txnbuild.TransactionParams{
+			SourceAccount: &channelAccount,
+			Operations:    originalTx.Operations(),
+			BaseFee:       int64(t.BaseFee),
+			Preconditions: txnbuild.Preconditions{
+				TimeBounds: txnbuild.NewTimeout(10),
+			},
+			IncrementSequenceNum: true,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building transaction: %w", err)
+	}
+	tx, err = t.ChannelAccountSignatureClient.SignStellarTransaction(ctx, tx, channelAccountPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction with channel account: %w", err)
+	}
+	// Wrap the transaction in a fee bump tx, signed by the distribution account
+	distributionAccountPublicKey, err := t.DistributionAccountSignatureClient.GetAccountPublicKey(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting distribution account public key: %w", err)
+	}
+
+	feeBumpTx, err := txnbuild.NewFeeBumpTransaction(
+		txnbuild.FeeBumpTransactionParams{
+			Inner:      tx,
+			FeeAccount: distributionAccountPublicKey,
+			BaseFee:    int64(t.BaseFee),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building fee-bump transaction %w", err)
+	}
+
+	feeBumpTx, err = t.DistributionAccountSignatureClient.SignStellarFeeBumpTransaction(ctx, feeBumpTx)
+	if err != nil {
+		return nil, fmt.Errorf("signing the fee bump transaction with distribution account: %w", err)
+	}
+	return feeBumpTx, nil
+}
+
+func (t *transactionService) parseErrorResultXDR(errorResultXdr string) (tss.RPCTXCode, error) {
+
+	//errorResult := xdr.TransactionResult{}
+	unMarshallErr := "unable to unmarshal errorResultXdr: %s"
+	//err := errorResult.UnmarshalBinary([]byte(errorResultXdr))
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(errorResultXdr)
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf(unMarshallErr, errorResultXdr)
+	}
+	dec := xdr3.NewDecoder(strings.NewReader(string(decodedBytes)))
+	var errorResult xdr.TransactionResult
+	_, err = dec.Decode(&errorResult)
```

Comment:
Seems `xdr3.NewDecoder` mentions in its doc: `Typically, Unmarshal should be used instead of manually creating a Decoder.`

Also, we can use `bytes.NewReader` here instead of parsing to string.

```suggestion
	var errorResult xdr.TransactionResult
	_, err = xdr3.Unmarshal(bytes.NewReader(decodedBytes), &errorResult)
```

Reply:
Done